src/aggregated_range_proof/dealer.rs

```diff
@@ -0,0 +1,174 @@
+use curve25519_dalek::ristretto::RistrettoPoint;
+use curve25519_dalek::scalar::Scalar;
+use curve25519_dalek::traits::Identity;
+use generators::GeneratorsView;
+use inner_product_proof;
+use proof_transcript::ProofTranscript;
+use std::clone::Clone;
+use util;
+
+use super::messages::*;
+
+/// Dealer is an entry-point API for setting up a dealer
+pub struct Dealer {}
+
+impl Dealer {
+    /// Creates a new dealer with the given parties and a number of bits
+    pub fn new(
+        transcript: &mut ProofTranscript,
+        n: usize,
+        m: usize,
+    ) -> Result<DealerAwaitingValues, &'static str> {
+        transcript.commit_u64(n as u64);
+        transcript.commit_u64(m as u64);
+        Ok(DealerAwaitingValues { n, m })
+    }
+}
+
+/// When the dealer is initialized, it only knows the size of the set.
+pub struct DealerAwaitingValues {
+    n: usize,
+    m: usize,
+}
+
+impl DealerAwaitingValues {
+    /// Combines commitments and computes challenge variables.
+    pub fn receive_value_commitments(
+        self,
+        transcript: &mut ProofTranscript,
+        vc: &Vec<ValueCommitment>,
+    ) -> (DealerAwaitingPoly, ValueChallenge) {
+        // TODO: test that vc is length `m`.
+        let mut A = RistrettoPoint::identity();
+        let mut S = RistrettoPoint::identity();
+
+        for commitment in vc.iter() {
+            // Commit each V individually
+            transcript.commit(commitment.V.compress().as_bytes());
+
+            // Commit sums of As and Ss.
+            A += commitment.A;
+            S += commitment.S;
```

Comment:
Why do we commit only sums of `A`s and `S`s? Is it because only the sums are available to the verifier?

Is it possible for colluding parties to form `A`s and `S`s that would cancel each other in the sum?

Reply:
I think the answer is maybe "no", because in order for the aggregated proof to verify, the `A`s and `S`s have to be formed with respect to each party's generators, and those generators are supposed to be orthogonal. So colluding parties who formed `A`s and `S`s that were related to each other would have to know relations between the supposedly orthogonal generators.

Reply:
yes, it's because only the sums are available to the verifier. (we can commit to individual V's on the other hand because they are available to the verifier).

And yes, your guess/explanation makes sense